magnum_cluster_api/resources.py

```diff
@@ -362,6 +364,77 @@ def get_object(self) -> dict:
+            if cinder.is_enabled(self.cluster):
+                volume_snapshot_classes.append(
+                    {
+                        "name": "block-snapshot",
+                        "annotations": {
+                            "snapshot.storage.kubernetes.io/is-default-class": "true"
+                        },
+                        "driver": "cinder.csi.openstack.org",
+                        "deletionPolicy": "Delete",
+                    }
+                )
+
+            # Add Manila CSI volume snapshot class if enabled and share_network_id is specified.
+            if manila.is_enabled(self.cluster):
+                share_network_id = self.cluster.labels.get(
+                    "manila_csi_share_network_id"
+                )
+                if share_network_id:
+                    volume_snapshot_classes.append(
+                        {
+                            "name": "share-snapshot",
+                            "annotations": {
+                                "snapshot.storage.kubernetes.io/is-default-class": "true"
+                            },
+                            "driver": "nfs.manila.csi.openstack.org",
+                            "deletionPolicy": "Delete",
+                            "parameters": {
+                                "csi.storage.k8s.io/snapshotter-secret-name": "csi-manila-secrets",
+                                "csi.storage.k8s.io/snapshotter-secret-namespace": "kube-system",
+                            },
+                        }
+                    )
+
+            controller_values = {
+                "fullnameOverride": "snapshot-controller",
+                "volumeSnapshotClasses": volume_snapshot_classes,
+            }
+
+            # Override image repository if custom registry is configured
+            if repository:  # repository is the container_infra_prefix
+                repository_clean = repository.rstrip("/")
+                # When using a custom registry, the image is named 'csi-snapshot-controller'
+                controller_values["image"] = {
+                    "repository": f"{repository_clean}/csi-snapshot-controller"
+                }
+            # For default registry, use the default image name 'snapshot-controller' from values.yaml
+
+            data = {
+                **data,
+                **{
+                    "snapshot-controller-csi.yaml": helm.TemplateReleaseCommand(
+                        namespace="kube-system",
+                        release_name="snapshot-controller",
+                        chart_ref=os.path.join(
+                            pkg_resources.resource_filename(
+                                "magnum_cluster_api", "charts"
+                            ),
+                            "snapshot-controller-csi/",
+                        ),
+                        values={"controller": controller_values},
+                    )(repository=repository)
+                },
+            }
+
```

Comment:
A lot of this code has been moved to the Rust counterpart, so probably best to move forward with putting it there.

Reply:
Please show me where to start looking.